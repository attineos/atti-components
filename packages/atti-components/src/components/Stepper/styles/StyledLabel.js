import styled from 'styled-components'

const StyledLabel = styled.p`
  position: relative;
  font-size: ${({ theme }) => theme.components.stepper.fonts.fontSize};
  font-family: ${({ theme }) => theme.components.stepper.fonts.fontFamily};
  text-align: center;
  color: ${({ theme, middle }) =>
    middle ? theme.components.stepper.colors.midText : theme.components.stepper.colors.text};
  margin-top: ${({ theme }) => theme.components.stepper.sizes.marginTopLabel};
`

export default StyledLabel
