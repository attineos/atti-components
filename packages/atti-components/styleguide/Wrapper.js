import React from 'react'

import ThemeProvider from '../src/components/ThemeProvider'
import { defaultTheme } from '../src/themes'

export default ({ children }) => <ThemeProvider otherTheme={defaultTheme}>{children}</ThemeProvider>
