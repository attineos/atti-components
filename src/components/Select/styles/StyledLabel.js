import styled, { css } from 'styled-components'
import Label from '../../Label'

const spaces = css`
  margin-bottom: ${({ theme }) => theme.components.select.label.spaces.marginBottom};
`

const StyledLabel = styled(Label)`
  ${spaces};
  display: block;
`

export default StyledLabel
