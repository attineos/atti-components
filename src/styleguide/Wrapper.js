import React from 'react'
import PropTypes from 'prop-types'
import { remove, get } from 'lodash'

import Cookies from './helpers/Cookies'
import themes from './config/themes'

import ThemeProvider from '../components/ThemeProvider'

const updateWrappers = []

export const getWrappersToUpdate = () => updateWrappers

class Wrapper extends React.Component {
  static propTypes = {
    /**
     * The child node.
     */
    children: PropTypes.node.isRequired,
  }

  state = {
    selectedTheme: null,
  }

  componentWillMount() {
    updateWrappers.push(this)

    const selectedTheme = Cookies.hasItem('selectedTheme')
      ? Cookies.getItem('selectedTheme')
      : 'default'

    this.state.selectedTheme = get(themes, selectedTheme, null)
  }

  componentWillUnmount() {
    remove(updateWrappers, item => item === this)
  }

  handleSetTheme = newTheme =>
    this.setState(
      {
        selectedTheme: newTheme,
      },
      () => this.forceUpdate(),
    )

  render() {
    const { children } = this.props
    const { selectedTheme } = this.state

    return (
      <ThemeProvider theme={selectedTheme} key={Math.random()}>
        {children}
      </ThemeProvider>
    )
  }
}

export default Wrapper
