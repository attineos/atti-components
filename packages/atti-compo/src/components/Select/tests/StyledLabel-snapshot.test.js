/* This file was generated automatically, please do not write in it. */
import React from 'react';
import StyledLabel from '../styles/StyledLabel';
import Wrapper from 'styleguide/Wrapper'
import renderer from 'react-test-renderer';

const mock = () => {}

it('StyledLabel renders correctly', () => {
  const tree = renderer
    .create(
      <Wrapper>
        <StyledLabel ></StyledLabel>
      </Wrapper>
      )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
    