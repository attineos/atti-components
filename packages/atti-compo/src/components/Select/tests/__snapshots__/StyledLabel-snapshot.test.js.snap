// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StyledLabel renders correctly 1`] = `
<label
  className="StyledLabel-sc-7lrvk2-0 bVnWMj Label-xhjve5-0 fQVHlM"
/>
`;
